### PR TITLE
CLOUDP-259458: Sort tags alphabetically

### PR DESCRIPTION
**Pull request by blva**
## Proposed changes

_Jira ticket:_ [CLOUDP-259458](https://jira.mongodb.org/browse/CLOUDP-259458)

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: tools/cli/internal/openapi/oasdiff.go [hunk 1]
```diff
@@ -233,11 +233,21 @@ func (o OasDiff) mergeTags() error {
 			}
 		}
 	}
-
+	sortTagsAlphabetically(baseTags)
 	o.base.Spec.Tags = baseTags
 	return nil
 }
 
+func sortTagsAlphabetically(tags openapi3.Tags) {
```
Comment (gssbzn) [inline tools/cli/internal/openapi/oasdiff.go:241]: https://pkg.go.dev/sort

File: tools/cli/internal/openapi/oasdiff_test.go [hunk 2]
```diff
@@ -305,6 +335,14 @@ func TestOasDiff_mergeTags(t *testing.T) {
...
 			}
 			tc.wantErr(t, o.mergeTags())
+			if tc.expectedTags != nil {
+				a := assert.New(t)
+				a.Equal(len(o.base.Spec.Tags), len(tc.expectedTags))
+				// check tag names are the same
+				for i := range o.base.Spec.Tags {
+					a.Equal(o.base.Spec.Tags[i].Name, tc.expectedTags[i].Name)
+				}
+			}
```
Comment (gssbzn) [inline tools/cli/internal/openapi/oasdiff_test.go:345]: wouldn't https://pkg.go.dev/github.com/stretchr/testify/assert#Equal work here?

Comment (gssbzn) [reply tools/cli/internal/openapi/oasdiff_test.go:345]: to eb clear assert.Equal(expectedTags, o.base.Spec.Tags)

File: tools/cli/internal/openapi/oasdiff_test.go [hunk 3]
```diff
@@ -305,6 +335,13 @@ func TestOasDiff_mergeTags(t *testing.T) {
 				external: tc.inputExternal,
 			}
 			tc.wantErr(t, o.mergeTags())
+			if tc.expectedTags != nil {
+				assert.Equal(t, len(o.base.Spec.Tags), len(tc.expectedTags))
+				// check tag names are the same
+				for i := range o.base.Spec.Tags {
+					assert.Equal(t, o.base.Spec.Tags[i].Name, tc.expectedTags[i].Name)
+				}
```
Comment (gssbzn) [inline tools/cli/internal/openapi/oasdiff_test.go:343]: wouldn't the following do the same? it should account for order
```suggestion
				assert.Equal(t, tc.expectedTags, o.base.Spec.Tags))
```

**Review by gssbzn — APPROVED** [review]
LGTM

File: tools/cli/internal/openapi/oasdiff_test.go (hunk 3 above)
Comment (blva) [reply tools/cli/internal/openapi/oasdiff_test.go:343]: no it doesn't work as it is a list of objects, it complains about the addresses

**Pull request merged**